indra/statements/validate.py

```diff
@@ -70,7 +80,71 @@ class InvalidStatement(ValueError):
     pass
 
 
-def validate_ns(db_ns):
+class IdentifiersValidator:
+    """A class that can be used to validate INDRA Statements."""
+
+    @staticmethod
+    def assert_valid_ns(db_ns):
+        if db_ns in non_grounding or db_ns in identifiers.non_registry:
+            return
+        identifiers_ns = \
+            identifiers.identifiers_mappings.get(db_ns, db_ns.lower())
+        if identifiers_ns in identifiers.identifiers_registry:
+            return
+        raise UnknownNamespace(db_ns)
+
+    @staticmethod
+    def assert_valid_id(db_ns, db_id):
+        if db_id is None:
+            raise MissingIdentifier(db_ns, None)
+        if db_ns in non_grounding or db_ns in identifiers.non_registry:
+            return
+        identifiers_ns = \
+            identifiers.identifiers_mappings.get(db_ns, db_ns.lower())
+        if identifiers_ns in identifiers.identifiers_registry:
+            pattern = identifiers.identifiers_registry[
+                identifiers_ns]['pattern_compiled']
+            if pattern.match(db_id):
+                return
+            else:
+                raise InvalidIdentifier(db_ns, db_id, pattern.pattern)
+        else:
+            raise UnknownIdentifier(db_ns, db_id)
+
+
+class BioregistryValidator:
+    """A class that can be used to validate INDRA Statements."""
+
+    @staticmethod
+    def assert_valid_ns(db_ns):
+        if db_ns in non_grounding:
+            return
+        prefix = bioregistry.bioregistry_overrides.get(db_ns, db_ns.lower())
+        if prefix in bioregistry.registry:
+            return
+        raise UnknownNamespace(db_ns)
+
+    @staticmethod
+    def assert_valid_id(db_ns, db_id):
+        if db_id is None:
+            raise MissingIdentifier(db_ns, None)
+        if db_ns in non_grounding:
+            return
+        prefix = bioregistry.bioregistry_overrides.get(db_ns, db_ns.lower())
+        if prefix in bioregistry.registry:
+            pattern = bioregistry.registry[prefix].get('pattern_compiled')
+            if not pattern or pattern.match(db_id):
```

Comment:
is there a practical difference between `pattern.match` and `pattern.fullmatch`?